### PR TITLE
Fetch exercises list

### LINKED ISSUE #275: Re-add option --id to learn-ocaml-client-fetch
In the current version of learn-ocaml-client there is no --id option anymore for the fetch command. I think this option is useful if the user have 5 exercises and wants just to download 1 file.

### PR DESCRIPTION
**Pull request by nobrakal**
Hi,

This fixes #275. It adds the capability to the client to `fetch` only some saves. The syntax looks like:

```
SYNOPSIS
       learn-ocaml-client fetch [OPTION]... [EXERCISE_ID]...

DESCRIPTION
       Fetch the user's solutions on the server to the current directory

ARGUMENTS
       EXERCISE_ID
           Exercise identifier. Can be repeated. If not present, all the
           exercises will be downloaded
```

### CODE REVIEW

File: src/main/learnocaml_client.ml [hunk 1]
```diff
@@ -154,6 +154,15 @@ module Args_exercises = struct
           $color_when $verbose )
 end
 
+module Args_fetch = struct
+  let id =
+    value & pos_all string [] & info [] ~docv:"EXERCISE_ID" ~doc:
+      "Exercise identifier. Can be repeated. \
+       If not present, all the exercises will be downloaded"
```
Comment (yurug) [inline src/main/learnocaml_client.ml]: Please add a final dot.

File: src/main/learnocaml_client.ml [hunk 2]
```diff
@@ -731,31 +740,50 @@ module Fetch = struct
            (Token.to_string token)
       | e -> Lwt.fail e
 
-  let write_save_files save =
-  Lwt_list.iter_s (fun (id, st) ->
+  let write_save_files lst save =
+    let pred x = match lst with
```
Comment (yurug) [inline src/main/learnocaml_client.ml]: This `pred` function seems weird. Can you describe its purpose?

File: src/main/learnocaml_client.ml (hunk 1 above)
Comment (nobrakal) [reply src/main/learnocaml_client.ml]: Speaking of it, there is no consensus in the code, some of the descriptions have a final dot (`--version`, `--help`), but not the others (`--server`, `--token`, for example).

File: src/main/learnocaml_client.ml (hunk 2 above)
Comment (nobrakal) [reply src/main/learnocaml_client.ml]: I am not good at choosing names :) It is a predicate to decide if we have to get the file or not. If no files where specified (hence an empty `lst`), we have to get all files, so the predicate is always true.

**Comment by nobrakal** [comment]
Oh, while I am at it, I will fix #278 , please don't merge this for now.

File: src/main/learnocaml_client.ml (hunk 1 above)
Comment (yurug) [reply src/main/learnocaml_client.ml]: I prefer a final dot everywhere then.

File: src/main/learnocaml_client.ml (hunk 2 above)
Comment (yurug) [reply src/main/learnocaml_client.ml]: Then, we need a comment to explain this.

**Comment by yurug** [comment]
Are you done with this PR, @nobrakal ?

**Comment by nobrakal** [comment]
Oops, yes indeed, all is good for me !

**Pull request merged**